src/Development/IDE/Spans/Calculate.hs

```diff
@@ -76,6 +79,14 @@ getSpanInfo mods tcm =
           | b `isSubspanOf` a = GT
           | otherwise         = compare (srcSpanStart a) (srcSpanStart b)
 
+-- | The locations in the typechecked module are as precise as the ones in the parsed module.
+-- Therefore we build up a map from OccName to the corresponding definition in the parsed module
+-- to lookup precise locations for things like multi-clause function definitions.
```

Comment:
This comment confuses me. I thought the issue was that the locations are messed up in the typechecked module and that's why you're using the parsed module instead.

Reply:
oops, my wording was broken there. Should be fixed now.

Reply:
:+1: Thanks!